app/models/provider_interface/provider_applications_page_state.rb

```diff
@@ -64,45 +44,55 @@ def status_filters
         offer_withdrawn
       ].map do |state_name|
         {
-          type: 'checkbox',
-          text: I18n.t!("provider_application_states.#{state_name}"),
-          name: state_name,
+          value: state_name,
+          label: I18n.t!("provider_application_states.#{state_name}"),
+          checked: applied_filters[:status]&.include?(state_name),
         }
       end
 
       {
-        heading: 'status',
-        input_config: status_options,
+        type: :checkboxes,
+        heading: 'Status',
+        name: 'status',
+        options: status_options,
       }
     end
 
-    def provider_filters_builder
-      input_config = ProviderOptionsService.new(provider_user).providers.map do |provider|
+    def provider_filter
+      provider_options = ProviderOptionsService.new(provider_user).providers.map do |provider|
         {
-          type: 'checkbox',
-          text: provider.name,
-          name: provider.id.to_s,
+          value: provider.id,
+          label: provider.name,
+          checked: applied_filters[:provider]&.include?(provider.id.to_s),
         }
       end
 
-      {
-        heading: 'provider',
-        input_config: input_config,
+      provider_filter = {
+        type: :checkboxes,
+        heading: 'Provider',
+        name: 'provider',
+        options: provider_options,
       }
+
+      return provider_filter if provider_filter[:options].size > 1
```

Comment:
It might be friendlier to return as early as possible, ie when we first do the query for providers, rather than first building the hash and then throwing it away?

Reply:
Good point there could be a few of these so would make sense to make it a bit faster.

Reply:
Have refactored so that less work is done before the check 👍 - see: https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/2114/commits/9b9f81c50e5f69c30f6612a9f29f095ff19a5e2d